src/grizzly/services/webtransport/core.py

```diff
@@ -0,0 +1,126 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+import asyncio
+import os
+from asyncio import AbstractEventLoop
+from logging import getLogger
+from pathlib import Path
+from platform import system
+from threading import Thread
+from typing import Optional
+
+from aioquic.asyncio import serve  # type: ignore[attr-defined]
+from aioquic.h3.connection import H3_ALPN
+from aioquic.quic.configuration import QuicConfiguration
+
+from ..base import BaseService
+from .wpt_h3_server import webtransport_h3_server
+from .wpt_h3_server.webtransport_h3_server import (
+    SessionTicketStore,
+    WebTransportH3Protocol,
+    _connect_to_server,
+)
+
+# Override global _doc_root
+webtransport_h3_server._doc_root = os.path.join(  # pylint: disable=protected-access
+    os.path.dirname(os.path.abspath(__file__)),
+    "wpt_h3_server",
+    "handlers",
+)
+
+LOG = getLogger(__name__)
+
+
+class WebTransportServer(BaseService):
+    def __init__(self, port: int, cert: Path, key: Path) -> None:
+        """A WebTransport service.
+
+        Args:
+            port: The port on which to listen on.
+            cert: The path to the certificate file.
+            key: The path to the certificate's private key.
+        """
+        self._port = port
+        self._cert = cert
+        self._key = key
+
+        self._loop: Optional[AbstractEventLoop] = None
+        self._server_thread: Optional[Thread] = None
+        self._started: bool = False
+
+    @property
+    def location(self) -> str:
+        return "grz_webtransport_server"
+
+    @property
+    def port(self) -> int:
+        """The port on which the service is listening"""
+        return self._port
+
+    def url(self, _query: str) -> bytes:
+        """URL for Sapphire.set_dynamic_response
+
+        Args:
+            _query (str): Unused query string.
```

Comment:
`(str)` is no longer needed in the docs since we have type hints now (I see this in a few places).

Reply:
Fixed in https://github.com/MozillaSecurity/grizzly/pull/484/commits/1ca85822fed0b92d50c033786ce63e681379ae51